frontend/src/scenes/surveys/components/StackedBar.tsx

```diff
@@ -0,0 +1,88 @@
+import clsx from 'clsx'
+import { Tooltip } from 'lib/lemon-ui/Tooltip'
+import { humanFriendlyNumber } from 'lib/utils'
+
+const formatCount = (count: number, total: number): string => {
+    if ((count / total) * 100 < 3) {
+        return ''
+    }
+    return `${humanFriendlyNumber(count)}`
+}
+
+// Define a type for the color classes to ensure type safety
+type ColorClass = 'bg-brand-blue' | 'bg-warning' | 'bg-success' | 'bg-danger'
+
+export interface StackedBarSegment {
+    count: number
+    label: string
+    colorClass: ColorClass
+    tooltip?: string
+}
+
+export function StackedBar({ segments }: { segments: StackedBarSegment[] }): JSX.Element {
+    const total = segments.reduce((sum, segment) => sum + segment.count, 0)
```

Comment:
can total be 0 here? if so, division will fail below

Reply:
there's a check here for [total > 0](https://github.com/PostHog/posthog/pull/30787/files/9bf1f4eec8ca1f7e2822377a89a82a78add68fe2#diff-76721f0fe73bb91755305cd2bbd687d586ced0b7a1177f37535e57cbe9c78771R28). but i'll refactor reduce one level of nesting and make it clearer too